enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/FulltextUpdateApplier.java

```diff
@@ -149,7 +149,7 @@ private BinaryLatch enqueuePopulateIndex( WritableFulltext index, GraphDatabaseS
         {
             PartitionedIndexWriter indexWriter = index.getIndexWriter();
             String[] indexedPropertyKeys = index.properties().toArray( new String[0] );
-            try ( Transaction tx = db.beginTx() )
+            try ( Transaction ignore = db.beginTx( 10, TimeUnit.HOURS ) )
```

Comment:
Couldn't this applier be started after recovery instead? To have this timeout _should_ be fine, but it's better to avoid timeouts altogether and change ordering, if possible.